sycl/include/sycl/accessor.hpp

```diff
@@ -1145,6 +1145,33 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
       : impl({}, detail::InitializedVal<AdjustedDim, range>::template get<0>(),
              detail::InitializedVal<AdjustedDim, range>::template get<0>()) {}
 
+  // implicit conversion between const / non-const types for read only accessors
+  template <typename DataT_, int Dimensions_, access::mode AccessMode_,
+            access::target AccessTarget_, access::placeholder IsPlaceholder_,
+            typename PropertyListT_>
+  accessor(const accessor<DataT_, Dimensions_, AccessMode_, AccessTarget_,
+                          IsPlaceholder_, PropertyListT_> &other,
+           std::enable_if_t<(AccessMode_ == access_mode::read) &&
+                            !(std::is_same<DataT_, DataT>::value) &&
+                            std::is_same<std::remove_const_t<DataT_>,
+                                         std::remove_const_t<DataT>>::value> * =
+               nullptr)
```

Comment:
Would it cause problems if we moved this to a template argument rather than an additional argument to the ctor?

Reply:
thanks @steffenlarsen, it's done.